src/main/java/uk/gov/justice/probation/courtcasematcher/service/MatcherService.java

```diff
@@ -53,8 +62,38 @@ public Mono<Defendant> matchDefendant(Defendant defendant, Hearing hearing) {
                     log.error(throwable.getMessage());
                     telemetryService.trackOffenderMatchFailureEvent(defendant, hearing);
                 })
+                .doOnSuccess(matchResponse -> enrichWithMatchScore(hearing, defendant, matchResponse))
```

Comment:
Would a `map` be more appropriate here? A `do` is effectively a side-effect which doesn't feel right since we do want to modify the response for the next consumer. I'll admit I'm not sure what will happen to any errors thrown here either but they aren't being handled explicitly which leads me to think it may not be good...

Reply:
Taking a look at the implementation at line 70 - the Mono returned by `enrichWithMatchScore` is redundant because `doOnSuccess` expects a [Consumer](https://docs.oracle.com/javase/8/docs/api/java/util/function/Consumer.html?is-external=true)<? super [Throwable](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html?is-external=true)> onError rather than a `Function` 

i.e. it doesn't use the return value for anything, that doesn't seem right...

Reply:
done ta